cmd/providers.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	v1 "github.com/odpf/guardian/api/handler/v1"
+	pb "github.com/odpf/guardian/api/proto/odpf/guardian"
+	"github.com/odpf/guardian/domain"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+func providersCommand(c *config, adapter v1.ProtoAdapter) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "providers",
+		Short: "manage providers",
+	}
+
+	cmd.AddCommand(listProvidersCommand(c))
+	cmd.AddCommand(createProviderCommand(c, adapter))
+	cmd.AddCommand(updateProviderCommand(c, adapter))
+
+	return cmd
+}
+
+func listProvidersCommand(c *config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "list",
+		Short: "list providers",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), time.Second*2)
```

Comment:
I see these connection codes being used frequently, can we wrap it to a function?
e.g.
```
ctx, cancel := connectToHost(ctx)
```

Reply:
it's a tricky one, we're deferring `dialCancel()` and `conn.Close()` from here only. I tried to put them into a function`createClient()` that also returns a function that calls both `dialCancel()` and `conn.Close()` for cancellation but it's not possible